### DISCUSSION TITLE
Cannot connect to printer

### DISCUSSION THREAD

**Discussion by quanghoath** [tr3v3r/react-native-esc-pos-printer#179] [Q&A]
i use version 4.2.0 but it doesn't connect to the printer. it hangs at tryToConnectUntil.

Here is my code.
const printerInstance = new Printer({
target: "192.168.1.230",
deviceName: "TM-T82",
});
const res = await printerInstance.addQueueTask(async () => {
await Printer.tryToConnectUntil(
printerInstance,
(status) => status.online.statusCode === PrinterConstants.TRUE
);
await printerInstance.addText("DUDE!");
await printerInstance.addFeedLine();
await printerInstance.addCut();
const result = await printerInstance.sendData();
await printerInstance.disconnect();
return result;
});

**Comment by aargon007** [comment]
@quanghoath do you connect the right device? target should be tcp/bt:mac address

**Reply by quanghoath** [reply]
<img width="1168" alt="Screenshot 2024-08-29 at 14 05 46" src="https://github.com/user-attachments/assets/f6357bd9-f638-43fa-abd4-1bfcbc59869f">

**Reply by quanghoath** [reply]
Printer information

**Reply by quanghoath** [reply]
<img width="959" alt="Screenshot 2024-08-29 at 14 09 40" src="https://github.com/user-attachments/assets/6155285d-ad73-438c-b684-58d6ee606ab4">
